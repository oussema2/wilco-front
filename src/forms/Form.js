import MobxReactForm from 'mobx-react-form';
import dvr from '../plugins/dvr';

export default class Form extends MobxReactForm {
	constructor(
		{ fields = [] } = {},
		{
			hooks,
			options = {
				validateOnChange: true,
				showErrorsOnChange: false,
				validationDebounceWait: 0
			},
			plugins = dvr
		} = {}
	) {
		super( { fields }, { options, hooks, plugins } );
	}
}
