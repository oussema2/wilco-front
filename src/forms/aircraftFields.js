import { MAKE_AND_MODEL, TAIL_NUMBER } from '../constants/formFields/aircraftForm';

const fields = [ {
	name: MAKE_AND_MODEL,
	label: 'Make and model',
	placeholder: 'e.g. Cirrus SR 22',
	rules: 'required'
}, {
	name: TAIL_NUMBER,
	label: 'Tail number',
	placeholder: 'e.g. N2129J'
} ];

export default fields;
