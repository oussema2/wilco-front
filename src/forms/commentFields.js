import { TEXT } from '../constants/formFields/commentForm';

const fields = [ {
	name: TEXT,
	label: 'Comment',
	placeholder: ''
} ];

export default fields;
