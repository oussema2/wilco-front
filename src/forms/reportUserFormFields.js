export const REPORT_REASON_FIELD = 'reason';
export const ADDITIONAL_INFORMATION_FIELD = 'details';

export const FIELDS = [
	{
		name: REPORT_REASON_FIELD,
		rules: 'required'
	},
	{
		name: ADDITIONAL_INFORMATION_FIELD,
		placeholder: 'Type your message here',
		rules: 'between:0,300'
	}
];
