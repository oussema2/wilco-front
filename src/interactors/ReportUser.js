export default class ReportUser {
	constructor( { pilotService } ) {
		this.pilotService = pilotService;
	}

	execute( { attributes } ) {
		return this
			.pilotService
			.reportUser( attributes );
	}
}
