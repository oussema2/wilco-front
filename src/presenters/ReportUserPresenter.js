import { makeAutoObservable, runInAction } from 'mobx';
import Form from '../forms/Form';
import { FIELDS } from '../forms/reportUserFormFields';

export const REPORT_REASON_OPTIONS = [
	{ label: 'They used inappropriate language', value: 'inappropriate_language' },
	{ label: 'They shared inappropriate content', value: 'inappropriate_content' },
	{ label: 'They are spamming me', value: 'spam' },
	{ label: 'Their profile is fake', value: 'fake_profile' },
	{ label: 'Something else', value: 'other' }
];

export default class ReportUserPresenter {
	isSendingReport = false;

	cancelButton = {
		text: 'Cancel',
		onPress: () => {},
		style: 'cancel'
	};

	confirmButton = {
		text: 'Yes',
		onPress: () => { this.navigationService.goBack(); }
	};

	constructor( {
		alertMessagesService,
		snackbarService,
		getCurrentPilot,
		reportUser,
		navigationService
	} ) {
		this.alertMessagesService = alertMessagesService;
		this.snackbarService = snackbarService;
		this.getCurrentPilot = getCurrentPilot;
		this.reportUser = reportUser;
		this.navigationService = navigationService;

		this.form = new Form(
			{ fields: FIELDS },
			{ options: { showErrorsOnChange: false, validateOnChange: true, validationDebounceWait: 0 } }
		);

		makeAutoObservable( this );
	}

	// eslint-disable-next-line class-methods-use-this
	get reportReasonOptions() {
		return REPORT_REASON_OPTIONS;
	}

	get sendButtonEnabled() {
		return !this.isSendingReport && this.form.isValid;
	}

	get sendButtonLabel() {
		return this.isSendingReport ? 'Sending...' : 'Send';
	}

	onCrossPressed = () => {
		this.alertMessagesService.showConfirmationAlert( {
			title: 'Report User',
			message: 'Are you sure you want to close?',
			cancelButton: this.cancelButton,
			confirmButton: this.confirmButton
		} );
	}

	onReportSent = ( pilotId ) => {
		this.isSendingReport = true;
		const { id: currentPilotId } = this.getCurrentPilot.execute();

		const attributes = {
			reported_pilot_id: pilotId,
			reporting_pilot_id: currentPilotId,
			...this.form.values()
		};

		return this
			.reportUser
			.execute( { attributes } )
			.then( () => {
				this.snackbarService.showSuccess( { message: 'Your report has been sent' } );
			} )
			.catch( () => {
				this.snackbarService.showError( { message: 'Your report hasn\'t been sent' } );
			} )
			.finally( () => {
				runInAction( () => {
					this.isSendingReport = false;
				} );
				this.navigationService.goBack();
			} );
	}
}
