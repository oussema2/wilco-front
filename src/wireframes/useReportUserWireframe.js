import { useCallback, useEffect, useState } from 'react';
import { BackHandler } from 'react-native';
import { useFocusEffect, useNavigation } from '@react-navigation/native';
import Api from '../services/Api';
import { useRootStore } from '../providers/RootStoreProvider';

import EntityServiceFactory from '../services/EntityServiceFactory';
import GetCurrentPilotFromStore from '../interactors/GetCurrentPilotFromStore';
import ReportUser from '../interactors/ReportUser';
import { useSnackbarService } from '../providers/SnackbarProvider';
import AlertMessagesService from '../services/AlertMessagesService';

import ReportUserPresenter from '../presenters/ReportUserPresenter';
import Pilot from '../entities/Pilot';

const useReportUserWireframe = () => {
	const [ presenterInstance, setPresenterInstance ] = useState( null );
	const rootStore = useRootStore();
	const { pilotStore, authenticationStore } = rootStore;
	const api = new Api( { authenticationStore } );
	const navigation = useNavigation();
	const snackbarService = useSnackbarService();

	useEffect( () => {
		const getCurrentPilot = new GetCurrentPilotFromStore( { store: pilotStore } );
		const pilotService = EntityServiceFactory.forPilots( { api, buildEntity: Pilot.fromJSON } );
		const reportUser = new ReportUser( { pilotService } );
		const alertMessagesService = AlertMessagesService.shared();

		const presenter = new ReportUserPresenter( {
			alertMessagesService,
			snackbarService,
			getCurrentPilot,
			reportUser,
			navigationService: navigation
		} );

		setPresenterInstance( presenter );
	}, [] );

	useFocusEffect(
		useCallback( () => {
			BackHandler.addEventListener( 'hardwareBackPress', presenterInstance?.onCrossPressed );

			return () => BackHandler.removeEventListener( 'hardwareBackPress', presenterInstance?.onCrossPressed );
		}, [ presenterInstance ] )
	);

	return presenterInstance;
};

export default useReportUserWireframe;
