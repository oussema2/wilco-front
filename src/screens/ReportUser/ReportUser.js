/* eslint-disable react/jsx-no-bind */
import React, { useRef } from 'react';
import { StatusBar, Text, View } from 'react-native';
import { observer } from 'mobx-react-lite';
import PropTypes from 'prop-types';

import BaseScreen from '../BaseScreen/BaseScreen';
import { KeyboardAwareScrollView } from '../../components/KeyboardAwareScrollView';
import { NavigationBar } from '../../components/NavigationBar';
import { HorizontalPadding } from '../../components/HorizontalPadding';
import TextArea from '../../components/TextArea/TextArea';
import { cross } from '../../assets/icons';

import useReportUserWireframe from '../../wireframes/useReportUserWireframe';
import { REPORT_REASON_FIELD, ADDITIONAL_INFORMATION_FIELD } from '../../forms/reportUserFormFields';

import styles from './ReportUser.styles';
import { PrimaryButton } from '../../components';
import { RadioList } from '../../components/RadioList';

const ReportUser = ( { route } ) => {
	const { pilotId } = route.params;
	const presenter = useReportUserWireframe();

	const textAreaRef = useRef( null );

	const _fixedBottomContent = () => (
		<View style={styles.buttonContainer}>
			<PrimaryButton
				testID="send-report"
				title={presenter.sendButtonLabel}
				size="big"
				disabled={!presenter.sendButtonEnabled}
				onPress={() => presenter.onReportSent( pilotId )}
			/>
		</View>
	);

	return ( !!presenter && (
		<BaseScreen isLoading={presenter?.isSendingReport}>
			<StatusBar barStyle="dark-content" />
			<HorizontalPadding>
				<NavigationBar
					title="Report User"
					rightAction={{
						image: cross,
						imageStyle: { width: 16, height: 16 },
						onPress: presenter.onCrossPressed
					}}
				/>
			</HorizontalPadding>
			<KeyboardAwareScrollView
				showsVerticalScrollIndicator={false}
				scrollToInputAdditionalOffset={150}
				getTextInputRefs={() => [ textAreaRef.current ]}
				alwaysBounceVertical={false}
				fixedBottomContent={_fixedBottomContent()}
			>
				<View style={styles.subHeaderContainer}>
					<Text style={styles.subHeaderText}>
						Tell us what happened
					</Text>
				</View>
				<View style={styles.ReportUserContent}>
					<HorizontalPadding style={{ paddingBottom: 20 }}>
						<RadioList
							testID="report-reasons-list"
							onChange={presenter.form.$( REPORT_REASON_FIELD ).bind().onChange}
							value={presenter.form.$( REPORT_REASON_FIELD ).value}
							options={presenter.reportReasonOptions}
						/>
						<TextArea
							testID="report-additional-information"
							maxLength={300}
							error={presenter.form.$( ADDITIONAL_INFORMATION_FIELD ).error}
							inputProps={{
								inputRef: textAreaRef,
								...presenter.form.$( ADDITIONAL_INFORMATION_FIELD ).bind()
							}}
						/>
					</HorizontalPadding>
				</View>
			</KeyboardAwareScrollView>
		</BaseScreen>
	) );
};

ReportUser.propTypes = {
	route: PropTypes.any.isRequired
};

export default observer( ReportUser );
