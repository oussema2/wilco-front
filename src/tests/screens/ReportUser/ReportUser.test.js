import React from 'react';
import { render } from '@testing-library/react-native';

import ReportUser from '../../../screens/ReportUser/ReportUser';
import Form from '../../../forms/Form';
import { FIELDS } from '../../../forms/reportUserFormFields';

import * as useReportUserWireframe from '../../../wireframes/useReportUserWireframe';
import mockUseView from '../../mocks/mockUseView';
import { REPORT_REASON_OPTIONS } from '../../../presenters/ReportUserPresenter';

jest.mock( '@react-navigation/native', () => ( {
	useNavigation: () => ( {} )
} ) );

describe( 'ReportUser', () => {
	const renderScreen = ( route ) => {
		mockUseView(
			useReportUserWireframe,
			{
				form: new Form(
					{ fields: FIELDS },
					{
						options: {
							showErrorsOnChange: false,
							validateOnChange: true,
							validationDebounceWait: 0
						}
					}
				),
				reportReasonOptions: REPORT_REASON_OPTIONS
			}
		);

		return render( <ReportUser route={route} /> );
	};

	test( 'renders the report options, a text area and a send button', () => {
		const route = { params: { userID: 123 } };
		const { toJSON } = renderScreen( route );

		expect( toJSON() ).toMatchSnapshot();
	} );
} );
