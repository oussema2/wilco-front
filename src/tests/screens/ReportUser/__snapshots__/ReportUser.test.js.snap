// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportUser renders the report options, a text area and a send button 1`] = `
<BVLinearGradient
  colors={
    Array [
      4291295231,
      4294967295,
    ]
  }
  endPoint={
    Object {
      "x": 0.25,
      "y": 0.3,
    }
  }
  locations={
    Array [
      0.5,
      1,
    ]
  }
  startPoint={
    Object {
      "x": 0,
      "y": 0,
    }
  }
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 5,
      "display": "flex",
      "flexDirection": "column",
      "height": "100%",
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="base-screen"
>
  <RNCSafeAreaView
    edges={
      Array [
        "top",
        "right",
        "bottom",
        "left",
      ]
    }
    style={
      Object {
        "height": "100%",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "paddingHorizontal": 20,
          },
          null,
        ]
      }
      testID="horizontalPadding-component"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "display": "flex",
            "flexDirection": "row",
            "justifyContent": "center",
            "minHeight": 44,
          }
        }
        testID="navigationBar-component"
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.25,
                "justifyContent": "flex-start",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "width": 20,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.5,
                "justifyContent": "center",
              },
            ]
          }
        >
          <Text
            onPress={null}
            style={
              Object {
                "alignItems": "center",
                "color": "#000000",
                "flex": 1,
                "fontFamily": "NotoSansJP-Black",
                "fontSize": 16,
                "fontWeight": undefined,
                "letterSpacing": -0.3,
                "lineHeight": 22,
                "textAlign": "center",
              }
            }
            testID="title-text"
          >
            Report User
          </Text>
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "display": "flex",
                "flexDirection": "row",
                "justifyContent": "center",
                "minHeight": 44,
              },
              Object {
                "flex": 0.25,
                "justifyContent": "flex-end",
              },
            ]
          }
        >
          <View
            style={
              Object {
                "width": 20,
              }
            }
            testID="rightPlaceholder-view"
          />
        </View>
      </View>
    </View>
    <View
      alwaysBounceVertical={false}
      getTextInputRefs={[Function]}
      onLayout={[Function]}
      scrollToInputAdditionalOffset={150}
      showsVerticalScrollIndicator={false}
      style={
        Array [
          Object {
            "flex": 1,
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <RCTScrollView
        contentContainerStyle={
          Object {
            "flexGrow": 1,
          }
        }
        keyboardShouldPersistTaps="handled"
        style={Object {}}
      >
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "height": 80,
                "justifyContent": "center",
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#767676",
                  "fontFamily": "NotoSansJP-Regular",
                  "fontSize": 14,
                  "fontWeight": undefined,
                  "letterSpacing": 0,
                  "lineHeight": 20,
                  "textAlign": "center",
                }
              }
            >
              Tell us what happened
            </Text>
          </View>
          <View>
            <View
              style={
                Array [
                  Object {
                    "paddingHorizontal": 20,
                  },
                  Object {
                    "paddingBottom": 20,
                  },
                ]
              }
              testID="horizontalPadding-component"
            >
              <RCTScrollView
                data={
                  Array [
                    Object {
                      "label": "They used inappropriate language",
                      "value": "inappropriate_language",
                    },
                    Object {
                      "label": "They shared inappropriate content",
                      "value": "inappropriate_content",
                    },
                    Object {
                      "label": "They are spamming me",
                      "value": "spam",
                    },
                    Object {
                      "label": "Their profile is fake",
                      "value": "fake_profile",
                    },
                    Object {
                      "label": "Something else",
                      "value": "other",
                    },
                  ]
                }
                getItem={[Function]}
                getItemCount={[Function]}
                keyExtractor={[Function]}
                onContentSizeChange={[Function]}
                onLayout={[Function]}
                onMomentumScrollBegin={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                style={
                  Object {
                    "borderBottomWidth": 0,
                    "borderWidth": 0,
                  }
                }
                testID="report-reasons-list"
                viewabilityConfigCallbackPairs={Array []}
              >
                <View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="row-inappropriate_language-testID"
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0,
                            "borderWidth": 0,
                            "flex": 1,
                            "flexDirection": "row",
                            "marginHorizontal": 0,
                            "marginVertical": 20,
                            "paddingHorizontal": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "overflow": "hidden",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "marginRight": 12,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#A0A0A0",
                                      "fontSize": 24,
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                      "fontWeight": 100,
                                    },
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                                testID="iconIcon"
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View>
                          <Text
                            numberOfLines={1}
                            style={
                              Object {
                                "fontFamily": "NotoSansJP-Regular",
                                "fontSize": 14,
                                "fontWeight": undefined,
                                "letterSpacing": 0,
                                "lineHeight": 20,
                              }
                            }
                          >
                            They used inappropriate language
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="row-inappropriate_content-testID"
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0,
                            "borderWidth": 0,
                            "flex": 1,
                            "flexDirection": "row",
                            "marginHorizontal": 0,
                            "marginVertical": 20,
                            "paddingHorizontal": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "overflow": "hidden",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "marginRight": 12,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#A0A0A0",
                                      "fontSize": 24,
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                      "fontWeight": 100,
                                    },
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                                testID="iconIcon"
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View>
                          <Text
                            numberOfLines={1}
                            style={
                              Object {
                                "fontFamily": "NotoSansJP-Regular",
                                "fontSize": 14,
                                "fontWeight": undefined,
                                "letterSpacing": 0,
                                "lineHeight": 20,
                              }
                            }
                          >
                            They shared inappropriate content
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="row-spam-testID"
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0,
                            "borderWidth": 0,
                            "flex": 1,
                            "flexDirection": "row",
                            "marginHorizontal": 0,
                            "marginVertical": 20,
                            "paddingHorizontal": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "overflow": "hidden",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "marginRight": 12,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#A0A0A0",
                                      "fontSize": 24,
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                      "fontWeight": 100,
                                    },
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                                testID="iconIcon"
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View>
                          <Text
                            numberOfLines={1}
                            style={
                              Object {
                                "fontFamily": "NotoSansJP-Regular",
                                "fontSize": 14,
                                "fontWeight": undefined,
                                "letterSpacing": 0,
                                "lineHeight": 20,
                              }
                            }
                          >
                            They are spamming me
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="row-fake_profile-testID"
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0,
                            "borderWidth": 0,
                            "flex": 1,
                            "flexDirection": "row",
                            "marginHorizontal": 0,
                            "marginVertical": 20,
                            "paddingHorizontal": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "overflow": "hidden",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "marginRight": 12,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#A0A0A0",
                                      "fontSize": 24,
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                      "fontWeight": 100,
                                    },
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                                testID="iconIcon"
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View>
                          <Text
                            numberOfLines={1}
                            style={
                              Object {
                                "fontFamily": "NotoSansJP-Regular",
                                "fontSize": 14,
                                "fontWeight": undefined,
                                "letterSpacing": 0,
                                "lineHeight": 20,
                              }
                            }
                          >
                            Their profile is fake
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      collapsable={false}
                      focusable={true}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onFocus={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                      testID="row-other-testID"
                    >
                      <View
                        style={
                          Object {
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0,
                            "borderWidth": 0,
                            "flex": 1,
                            "flexDirection": "row",
                            "marginHorizontal": 0,
                            "marginVertical": 20,
                            "paddingHorizontal": 0,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "overflow": "hidden",
                            }
                          }
                        >
                          <View
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "marginRight": 12,
                              }
                            }
                          >
                            <View
                              style={
                                Object {
                                  "alignItems": "center",
                                  "backgroundColor": "transparent",
                                  "justifyContent": "center",
                                }
                              }
                            >
                              <Text
                                allowFontScaling={false}
                                style={
                                  Array [
                                    Object {
                                      "color": "#A0A0A0",
                                      "fontSize": 24,
                                    },
                                    Object {
                                      "backgroundColor": "transparent",
                                      "fontWeight": 100,
                                    },
                                    Object {
                                      "fontFamily": "FontAwesome",
                                      "fontStyle": "normal",
                                      "fontWeight": "normal",
                                    },
                                    Object {},
                                  ]
                                }
                                testID="iconIcon"
                              >
                                
                              </Text>
                            </View>
                          </View>
                        </View>
                        <View>
                          <Text
                            numberOfLines={1}
                            style={
                              Object {
                                "fontFamily": "NotoSansJP-Regular",
                                "fontSize": 14,
                                "fontWeight": undefined,
                                "letterSpacing": 0,
                                "lineHeight": 20,
                              }
                            }
                          >
                            Something else
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
              <View
                style={
                  Object {
                    "minHeight": 98,
                  }
                }
                testID="report-additional-information"
              >
                <View
                  style={
                    Object {
                      "alignContent": "center",
                      "alignItems": "flex-start",
                      "display": "flex",
                      "flexDirection": "row",
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "color": "#242424",
                        "fontFamily": "NotoSansJP-Black",
                        "fontSize": 16,
                        "fontWeight": undefined,
                        "letterSpacing": -0.3,
                        "lineHeight": 22,
                        "marginBottom": 4,
                      }
                    }
                    testID="input-label"
                  >
                    
                  </Text>
                </View>
                <View
                  style={
                    Object {
                      "flex": 1,
                    }
                  }
                >
                  <TextInput
                    autoCapitalize="none"
                    editable={true}
                    multiline={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onFocus={[Function]}
                    placeholder="Type your message here"
                    placeholderTextColor="#A0A0A0"
                    scrollEnabled={false}
                    selectionColor="#12A3BE"
                    style={
                      Object {
                        "backgroundColor": "#F2F2F2",
                        "borderRadius": 8,
                        "color": "#000000",
                        "fontFamily": "NotoSansJP-Bold",
                        "fontSize": 14,
                        "fontWeight": "normal",
                        "letterSpacing": 0,
                        "lineHeight": 20,
                        "minHeight": 48,
                        "paddingBottom": 16,
                        "paddingHorizontal": 12,
                        "paddingTop": 16,
                        "textAlignVertical": "top",
                      }
                    }
                    testID="input"
                    value=""
                  />
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "display": "flex",
                        "flexDirection": "row",
                        "justifyContent": "flex-end",
                        "marginTop": 10,
                      }
                    }
                  >
                    <Text
                      align="right"
                      style={
                        Object {
                          "color": "#767676",
                          "fontFamily": "NotoSansJP-Regular",
                          "fontSize": 12,
                          "fontWeight": undefined,
                          "letterSpacing": 0,
                          "lineHeight": 16,
                        }
                      }
                    >
                      0
                    </Text>
                    <Text
                      style={
                        Object {
                          "color": "#767676",
                          "fontFamily": "NotoSansJP-Regular",
                          "fontSize": 12,
                          "fontWeight": undefined,
                          "letterSpacing": 0,
                          "lineHeight": 16,
                        }
                      }
                    >
                      /300
                    </Text>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </RCTScrollView>
      <View
        style={
          Object {
            "alignSelf": "center",
            "marginBottom": 8,
            "marginTop": 8,
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "borderRadius": 3,
                "overflow": "hidden",
              },
              Object {
                "borderRadius": 3,
              },
              undefined,
              false,
            ]
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "busy": false,
                "disabled": true,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID="send-report"
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "hsl(208, 8%, 90%)",
                  "borderColor": "#2089dc",
                  "borderRadius": 26,
                  "borderWidth": 0,
                  "flexDirection": "row",
                  "height": 60,
                  "justifyContent": "center",
                  "padding": 8,
                  "width": 335,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaView>
</BVLinearGradient>
`;
